src/threadpool.c

```diff
@@ -262,6 +279,143 @@ static void init_once(void) {
 }
 
 
+int uv__threads_spin(unsigned int n) {
+  uv_thread_options_t config;
+  int r;
+  size_t i;
+  uv_thread_t* threads_tmp;
+
+  assert(n > nthreads);
+
+  config.flags = UV_THREAD_HAS_STACK_SIZE;
+  config.stack_size = 8u << 20;  /* 8 MB */
+
+  threads_tmp = uv__malloc(n * sizeof(threads[0]));
+  if (threads_tmp == NULL) {
+    return UV_ENOMEM;
+  }
+
+  /* Copy all threads into the new list. */
+  for (i = 0; i < nthreads; i++)
+    threads_tmp[i] = threads[i];
+
+  /* Free the old list. And make it points to the new one */
+  if (threads != default_threads)
+    uv__free(threads);
+
+  threads = threads_tmp;
+
+  r = uv_sem_init(&sem, 0);
+  if (r)
+    goto out;
+
+  while (nthreads < n) {
+    /* Wire up the error code to the return value. */
+    r = uv_thread_create_ex(threads + nthreads, &config, worker, &sem);
+    if (r)
+      break;
+
+    uv_sem_wait(&sem);
+    nthreads++;
+  }
+
+  assert(nthreads == n);
```

Comment:
In case thread creation fails, the assertion would not hold, right? Probably fixed by replacing the `break` with a `goto out`.

Reply:
Right. Let me fix that one